### DISCUSSION TITLE
Unlimited turtle blocks

### DISCUSSION THREAD

**Discussion by pre-alpha-final** [cc-tweaked/CC-Tweaked#1594] [Q&A] [answered]
Building a turtle based 3d printer. The problem is that the turtle needs to go back to a chest to get more blocks once it runs out. Would be great if it could have infinite blocks just like a player (creative mode obviously). Any ideas welcome, including hacks like running a command to spawn a dropped block in front of the turtle and suck() (though turtles can't run commands I think).

**Comment by SammyForReal** [comment]
**As an hacky solution**:
You could use a command block computer and make it spawn the requested blocks as items at the turtle after it sends a message via an _(wireless)_ modem to said PC.

Of course this would be more than dangerous on a survival server without any encryption etc. For the protocol then tho.

**Reply by pre-alpha-final** [reply]
So the turtle wouldn't `place()` anything you mean? It would just be an anchor to a world position and the block itself would be created via the command. Kinda ruins the magic of the turtle actually building things. It does however allow for more than 16 types of blocks to be used (now capped by turtle inventory limit).

**Comment by Lupus590** [comment]
If you have an ender chest mod then you could have infinite item generation feeding the ender chest and have the turtle place, such from and break the ender chest every time it needs more items.

**Reply by pre-alpha-final** [reply]
It's an option but then feeding the turtle becomes a bigger problem then the printing itself. Would prefer to go with something more straightforward.

**Comment by SirEdvin** [comment, marked as answer]
Well, you can use [Creative chest](https://docs.siredvin.site/turtlematic/miscellaneous/creative_chest/) from [Turtlematic](https://modrinth.com/mod/turtlematic) addon

**Reply by SirEdvin** [reply]
But I would mark this as idea for my gamerule-based CC addon in future :)

**Reply by pre-alpha-final** [reply]
Wow, if this does what it says on the tin, it's definitely the solution I'm going for. Will test it out later. Shame not a part of cc-tweaked though.
